pkg/graveler/graveler.go

```diff
@@ -1684,31 +1712,43 @@ func (g *KVGraveler) Commit(ctx context.Context, repositoryID RepositoryID, bran
 		if params.SourceMetaRange != nil {
 			commit.MetaRangeID = *params.SourceMetaRange
 		} else {
-			changes, err := g.StagingManager.List(ctx, branch.StagingToken, ListingMaxBatchSize)
+			itrs, err := g.listSealedTokens(ctx, branch)
 			if err != nil {
-				return "", fmt.Errorf("staging list: %w", err)
+				return nil, err
+			}
+			if len(itrs) == 0 { // staging empty
+				return nil, ErrNotFound
+			}
```

Comment:
This is a funky situation, I would log it..

Reply:
Also - I don't think the error is appropriate.

Reply:
I deleted it, it is checked on the commit manager